packages/compartment-mapper/src/link.js

```diff
@@ -45,10 +45,8 @@ const defaultCompartment = Compartment;
 const q = JSON.stringify;
 
 /**
- * @template {string | number} Key
- * @template Value
```

Comment:
There laid your problem!

It should have been `@template {any} Value`

`Value` is a deprecated alias for `AmountValue`.

You should be able to keep it generic if you want.

Reply:
After double checking, adding a type constraint wouldn't have helped, only renaming to something that didn't conflict with a type in scope would have, e.g. `TValue`